201-encrypt-create-new-vm-gallery-image/azuredeploy.json

```diff
@@ -175,7 +175,7 @@
           "osDisk": {
             "name": "[variables('osDiskName')]",
             "vhd": {
-              "uri": "[concat('http://',parameters('newStorageAccountName'),'.blob.core.windows.net/',parameters('vmStorageContainerName'),'/',variables('osDiskName'),'.vhd')]"
+              "uri": "[concat('https://',parameters('newStorageAccountName'),'.blob.core.windows.net/',parameters('vmStorageContainerName'),'/',variables('osDiskName'),'.vhd')]"
```

Comment:
Best practice is to dynamically retrieve the endpoint rather than hardcode any part of the url. Something like this:

[concat(reference(resourceId('Microsoft.Storage/storageAccounts', variables('newStorageAccountName')), '2016-01-01').primaryEndpoints.blob, variables('vmStorageContainerName'), '/', variables('osDiskName'), '.vhd')]